rs/nns/handlers/root/impl/canister/canister.rs

```diff
@@ -72,20 +70,15 @@ fn new_management_canister_client() -> impl ManagementCanisterClient {
 // messages are quite obscure.
 #[export_name = "canister_init"]
 fn canister_init() {
-    dfn_core::printer::hook();
     println!("{}canister_init", LOG_PREFIX);
 }
 
-#[export_name = "canister_post_upgrade"]
+#[post_upgrade]
 fn canister_post_upgrade() {
-    dfn_core::printer::hook();
     println!("{}canister_post_upgrade", LOG_PREFIX);
-    // Wipe out stable memory, because earlier version of this canister were
-    // stateful. This minimizes risk of future misinterpretation of data.
-    stable::set(&[]);
 }
 
-ic_nervous_system_common_build_metadata::define_get_build_metadata_candid_method! {}
+ic_nervous_system_common_build_metadata::define_get_build_metadata_candid_method_cdk! {}
```

Comment:
Do we still need this, given that metadata sections are now expected to be encoded on top of the WASM?

Reply:
NCR for this PR

Reply:
seems like an orthogonal change, but I'm okay removing these from all our canisters.  I didn't want to make any changes outside of the scope of the migration here.